src/master.rs

```diff
@@ -1,13 +1,56 @@
 ///! LIN bus master implementation
+use checksum;
 use driver;
 
 pub trait Master {
     type Error;
     fn send_wakeup(&mut self) -> Result<(), Self::Error>;
-    fn write_frame(&mut self, pid: u8, data: &[u8]) -> Result<(), Self::Error>;
+    fn write_frame(&mut self, frame: &Frame) -> Result<(), Self::Error>;
     fn read_frame(&mut self, pid: u8, data: &mut [u8]) -> Result<(), Self::Error>;
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct Frame {
+    pid: u8,
+    buffer: [u8; 9],
+    data_length: usize,
+}
+
+impl Frame {
+    /// Creates a LIN frame from the PID and data. Calculates and adds checksum accordingly
+    pub fn from_data(pid: u8, data: &[u8]) -> Frame {
+        assert!(data.len() <= 8, "Maximum data is 8 bytes");
+        let mut buffer = [0u8; 9];
+        buffer[0..data.len()].clone_from_slice(data);
+        buffer[data.len()] = checksum(pid, &buffer[0..data.len()]);
+        Frame {
+            pid: pid,
+            buffer: buffer,
+            data_length: data.len(),
+        }
+    }
+
+    /// Access the data from the frame
+    pub fn get_data(&self) -> &[u8] {
+        &self.buffer[0..self.data_length]
+    }
+
+    /// Get the checksum from the frame
+    pub fn get_checksum(&self) -> u8 {
+        self.buffer[self.data_length]
+    }
+
+    /// Get the PID from the frame
+    pub fn get_pid(&self) -> u8 {
+        self.pid
+    }
+
+    /// Get the serialized bytes to write to the driver
+    pub fn get_data_with_checksum(&self) -> &[u8] {
+        &self.buffer[0..self.data_length + 1]
```

Comment:
why not use the inclusive operator `...`?  `&self.buffer[0...self.data_length]`

Reply:
It's actually `..=` but yes, since it finally [landed in stable 1.26](https://blog.rust-lang.org/2018/05/10/Rust-1.26.html#inclusive-ranges-with-) we can of course use it.

Reply:
ok, just saw that it is pretty new, so if you think we should leave it as it is, its ok with me

Reply:
I think we can use it, there aren't much reasons to support older Rust versions in such an early state. Done.